apps/dashboard/src/components/workflow-editor/steps/controls/text-widget.tsx

```diff
@@ -1,34 +1,64 @@
+import { useMemo } from 'react';
+import { type WidgetProps } from '@rjsf/utils';
+import { useFormContext } from 'react-hook-form';
+import { EditorView } from '@uiw/react-codemirror';
 import { Editor } from '@/components/primitives/editor';
 import { FormControl, FormField, FormItem, FormLabel, FormMessage } from '@/components/primitives/form/form';
-import { InputField } from '@/components/primitives/input';
+import { Input, InputField } from '@/components/primitives/input';
 import { completions } from '@/utils/liquid-autocomplete';
 import { capitalize } from '@/utils/string';
 import { autocompletion } from '@codemirror/autocomplete';
-import { type WidgetProps } from '@rjsf/utils';
-import { useFormContext } from 'react-hook-form';
+import { getFieldName } from './template-utils';
+import { parseStepVariablesToLiquidVariables } from '@/utils/parseStepVariablesToLiquidVariables';
+import { useStep } from '../step-provider';
 
 export function TextWidget(props: WidgetProps) {
-  const { label, readonly, name } = props;
+  const { label, readonly, disabled, id, required } = props;
   const { control } = useFormContext();
+  const { step } = useStep();
+  const variables = useMemo(() => (step ? parseStepVariablesToLiquidVariables(step.variables) : []), [step]);
+
+  const extractedName = useMemo(() => getFieldName(id), [id]);
+  const isNumberType = useMemo(() => props.schema.type === 'number', [props.schema.type]);
 
   return (
     <FormField
       control={control}
-      name={name}
+      name={extractedName}
       render={({ field }) => (
-        <FormItem className="my-2 py-1">
-          <FormLabel>{capitalize(label)}</FormLabel>
+        <FormItem className="w-full py-1">
+          <FormLabel className="text-xs">{capitalize(label)}</FormLabel>
           <FormControl>
-            <InputField size="fit">
-              <Editor
-                fontFamily="inherit"
-                placeholder={capitalize(label)}
-                id={label}
-                extensions={[autocompletion({ override: [completions([])] })]}
-                value={field.value}
-                onChange={(val) => field.onChange(val)}
-                readOnly={readonly}
-              />
+            <InputField size="fit" className="w-full">
+              {isNumberType ? (
```

Comment:
Everything is controlled by user via the schema
![image](https://github.com/user-attachments/assets/e7a08803-fa9e-4f02-a1bf-a02c6ba2050d)
but this also faces the same problem like the number field
user defines `emailField: z.email()` and then types the `payload` this will error out coz of zod schema validation